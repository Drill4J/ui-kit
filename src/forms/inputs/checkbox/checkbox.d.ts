import { CheckboxProps } from './checkbox-types';
export declare const Checkbox: ({ className, onChange, checked, label, value, color, }: CheckboxProps) => JSX.Element;
