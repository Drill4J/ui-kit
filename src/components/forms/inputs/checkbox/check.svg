<svg xmlns="http://www.w3.org/2000/svg" width="14" height="14" viewBox="0 0 16 16" version="1.1">
    <g stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <path d="M1.727 6.821a.935.935 0 0 0 .001 1.376l4.294 4.106a.99.99 0 0 0 1.417-.04l6.872-7.358c.38-.408.36-1.045-.038-1.417l.204.19a.982.982 0 0 0-1.398.057L7.42 9.932a.968.968 0 0 1-1.394.038L2.973 7.03a1.068 1.068 0 0 0-1.45-.018l.204-.19z" fill="#FFF" />
    </g>
</svg>