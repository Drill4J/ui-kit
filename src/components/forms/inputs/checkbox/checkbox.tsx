import { styled } from 'twin.macro';
// @ts-ignore
import icon from "./check.svg";

export const Checkbox = ({ field, ...props }: any) => (
    <Input type="checkbox" className={`
      appearance-none
      min-w-16px
      min-h-16px
      bg-monochrome-white 
      bg-no-repeat
      border rounded 
      border-current-color
      cursor-pointer
      hover:border-current-color
      checked:bg-current-color
      checked:hover:opacity-[0.8]
      checked:border-transparent
      disabled:opacity-50
      `}
      {...field}
      {...props} 
    />
);
const Input = styled.input`
  background-image: url(${icon});
  &:checked + svg {
    display: block;
  }
`;
