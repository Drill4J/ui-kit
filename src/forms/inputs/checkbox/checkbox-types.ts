export interface CheckboxProps {
  color?: string;
  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
  checked?: boolean;
  label?: string;
  disabled?: boolean;
  value?: string;
}
