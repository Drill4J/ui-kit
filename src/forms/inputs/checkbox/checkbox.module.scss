@import '../../../style-guide/index';

.checkbox {
  display: flex;
  align-items: center;
  color: map-get($monochrome, medium-tint);
  cursor: pointer;

  &__input {
    display: none;
  }

  &__icon-wrapper {
    display: flex;
    align-items: center;
    margin-left: 16px;
  }

  &__label {
    font-size: 14px;
    line-height: 24px;
    color: map-get($monochrome, black);
  }

  &--checked {
    color: map-get($primary-blue, default);
  }

  &--disabled {
    opacity: 0.5;
    pointer-events: none;
  }
}
