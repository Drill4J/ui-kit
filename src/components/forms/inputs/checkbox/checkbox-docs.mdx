import { Canvas, Meta, Story, ArgsTable } from '@storybook/addon-docs';

import { Checkbox } from './checkbox';

# Checkbox

The component gets its color from the wrapper component.

Example:

<Canvas>
  <Story 
    id="checkbox--checkbox-with-color"
    name="checkbox"
  />
</Canvas>
