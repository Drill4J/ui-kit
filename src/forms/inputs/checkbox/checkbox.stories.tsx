import { useState } from 'react';
import { storiesOf } from '@storybook/react';

import { Checkbox } from './checkbox';

storiesOf('Checkbox', module).add('Checkbox', () => {
  const [checkedFirst, setCheckedFirst] = useState(false);
  const [checkedSecond, setCheckedSecond] = useState(false);
  const [checkedThre, setCheckedThre] = useState(false);
  return (
    <div style={{ display: 'grid', gridTemplateColumns: 'auto auto auto' }}>
      <Checkbox
        checked={checkedFirst}
        label={checkedFirst ? 'checked' : 'not checked'}
        onChange={() => (checkedFirst ? setCheckedFirst(false) : setCheckedFirst(true))}
      />
      <Checkbox
        checked={checkedSecond}
        label={checkedSecond ? 'checked' : 'not checked'}
        onChange={() => (checkedSecond ? setCheckedSecond(false) : setCheckedSecond(true))}
      />
      <Checkbox
        checked={checkedThre}
        label="disable"
        onChange={() => (checkedThre ? setCheckedThre(false) : setCheckedThre(true))}
        disabled
      />
      <Checkbox
        checked={checkedFirst}
        label={checkedFirst ? 'checked' : 'not checked'}
        onChange={() => (checkedFirst ? setCheckedFirst(false) : setCheckedFirst(true))}
        color="blue"
      />
      <Checkbox
        checked={checkedSecond}
        label={checkedSecond ? 'checked' : 'not checked'}
        onChange={() => (checkedSecond ? setCheckedSecond(false) : setCheckedSecond(true))}
        color="blue"
      />
      <Checkbox
        checked={checkedThre}
        label="disable"
        onChange={() => (checkedThre ? setCheckedThre(false) : setCheckedThre(true))}
        color="blue"
        disabled
      />
      <Checkbox
        checked={checkedFirst}
        label={checkedFirst ? 'checked' : 'not checked'}
        onChange={() => (checkedFirst ? setCheckedFirst(false) : setCheckedFirst(true))}
        color="green"
      />
      <Checkbox
        checked={checkedSecond}
        label={checkedSecond ? 'checked' : 'not checked'}
        onChange={() => (checkedSecond ? setCheckedSecond(false) : setCheckedSecond(true))}
        color="green"
      />
      <Checkbox
        checked={checkedThre}
        label="disable"
        onChange={() => (checkedThre ? setCheckedThre(false) : setCheckedThre(true))}
        color="green"
        disabled
      />
      <Checkbox
        checked={checkedFirst}
        label={checkedFirst ? 'checked' : 'not checked'}
        onChange={() => (checkedFirst ? setCheckedFirst(false) : setCheckedFirst(true))}
        color="red"
      />
      <Checkbox
        checked={checkedSecond}
        label={checkedSecond ? 'checked' : 'not checked'}
        onChange={() => (checkedSecond ? setCheckedSecond(false) : setCheckedSecond(true))}
        color="red"
      />
      <Checkbox
        checked={checkedThre}
        label="disable"
        onChange={() => (checkedThre ? setCheckedThre(false) : setCheckedThre(true))}
        color="red"
        disabled
      />
      <Checkbox
        checked={checkedFirst}
        label={checkedFirst ? 'checked' : 'not checked'}
        onChange={() => (checkedFirst ? setCheckedFirst(false) : setCheckedFirst(true))}
        color="yellow"
      />
      <Checkbox
        checked={checkedSecond}
        label={checkedSecond ? 'checked' : 'not checked'}
        onChange={() => (checkedSecond ? setCheckedSecond(false) : setCheckedSecond(true))}
        color="yellow"
      />
      <Checkbox
        checked={checkedThre}
        label="disable"
        onChange={() => (checkedThre ? setCheckedThre(false) : setCheckedThre(true))}
        color="yellow"
        disabled
      />
      <Checkbox
        checked={checkedFirst}
        label={checkedFirst ? 'checked' : 'not checked'}
        onChange={() => (checkedFirst ? setCheckedFirst(false) : setCheckedFirst(true))}
        color="orange"
      />
      <Checkbox
        checked={checkedSecond}
        label={checkedSecond ? 'checked' : 'not checked'}
        onChange={() => (checkedSecond ? setCheckedSecond(false) : setCheckedSecond(true))}
        color="orange"
      />
      <Checkbox
        checked={checkedThre}
        label="disable"
        onChange={() => (checkedThre ? setCheckedThre(false) : setCheckedThre(true))}
        color="orange"
        disabled
      />
    </div>
  );
});
