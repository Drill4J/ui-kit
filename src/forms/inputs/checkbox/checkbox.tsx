import * as React from 'react';
import { BEM, input, div } from '@redneckz/react-bem-helper';

import { Icons } from '../../../components/icon';
import { CheckboxProps } from './checkbox-types';

import styles from './checkbox.module.scss';

const checkbox = BEM(styles);

export const Checkbox = checkbox(
  ({
    className, onChange, checked, label, value, color,
  }: CheckboxProps) => {
    const handleOnChange = () => {
      onChange &&
        onChange({
          target: {
            type: 'checkbox',
            checked: !checked,
          },
        } as React.ChangeEvent<HTMLInputElement>);
    };
    return (
      <div className={className} onClick={handleOnChange}>
        <CheckboxInput name={value} checked={checked} />
        <CheckboxIconWrapper type={!checked ? label : undefined} color={color || 'blue'}>
          {checked && <CheckMarkIcon width={10} height={7} />}
        </CheckboxIconWrapper>
        {label && <Label>{label}</Label>}
      </div>
    );
  },
);

const CheckboxInput = checkbox.checkboxInput(
  input({ type: 'checkbox', checked: false, name: '' } as {
    type?: string;
    checked?: boolean;
    name?: string;
  }),
);
const CheckboxIconWrapper = checkbox.checkboxIconWrapper(div(
  { color: 'blue' } as {color?: 'blue' | 'green' | 'red' | 'yellow' | 'orange'},
));
const CheckMarkIcon = checkbox.checkMarkIcon(Icons.Check);
const Label = checkbox.label('div');
