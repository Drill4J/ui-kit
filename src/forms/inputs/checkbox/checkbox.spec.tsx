import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import { Checkbox } from './checkbox';


describe('Checkbox', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(
      <Checkbox
        checked
        label="checkbox"
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should calls onChange when the checkbox label is clicked', () => {
    const wrapper = mount(<Checkbox label="Click me!" />);
    const checkbox = () => wrapper.find('input');
    expect(wrapper.props().label).toEqual('Click me!');
    checkbox().simulate('change', { target: { checked: true } });
  });
});
