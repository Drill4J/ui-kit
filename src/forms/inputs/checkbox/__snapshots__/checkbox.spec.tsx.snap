// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should match snapshot 1`] = `
<div
  className="sc-dPiKHq gcpgfU"
  onClick={[Function]}
>
  <input
    checked={true}
    className="sc-bBHHQT sQSgW"
    type="checkbox"
  />
  <div
    checked={true}
    className="sc-cNKpQo hVgyxZ"
    data-test="checkbox:input"
  >
    <svg
      className="sc-gsDJrp eoXRPA sc-AjmZR ixOclY"
      fillRule="evenodd"
      height={7}
      style={
        Object {
          "WebkitTransform": "rotate(0deg)",
        }
      }
      transform="rotate(0)"
      viewBox="0 0 14 10"
      width={10}
    >
      <path
        d="M.727 3.821a.935.935 0 0 0 .001 1.376l4.294 4.106a.99.99 0 0 0 1.417-.04l6.872-7.358c.38-.408.36-1.045-.038-1.417l.204.19a.982.982 0 0 0-1.398.057L6.42 6.932a.968.968 0 0 1-1.394.038L1.973 4.03a1.068 1.068 0 0 0-1.45-.018l.204-.19z"
      />
    </svg>
  </div>
  <div
    className="sc-khQdMy gAOLRn"
    data-test="checkbox:label"
  >
    checkbox
  </div>
</div>
`;
